go.mod

```diff
@@ -4,83 +4,352 @@ go 1.24.0
 
 require (
 	github.com/go-logr/logr v1.4.2
-	github.com/gobuffalo/flect v1.0.2
+	github.com/gobuffalo/flect v1.0.3
 	github.com/google/cel-go v0.24.1
-	github.com/onsi/ginkgo/v2 v2.20.0
-	github.com/onsi/gomega v1.34.1
-	github.com/prometheus/client_golang v1.19.1
-	github.com/spf13/cobra v1.8.1
-	github.com/stretchr/testify v1.9.0
-	go.uber.org/zap v1.26.0
-	golang.org/x/exp v0.0.0-20240719175910-8a7402abbf56
-	golang.org/x/time v0.3.0
-	google.golang.org/genproto/googleapis/api v0.0.0-20240826202546-f6391c0de4c7
-	k8s.io/api v0.31.0
-	k8s.io/apiextensions-apiserver v0.31.0
-	k8s.io/apimachinery v0.31.0
-	k8s.io/apiserver v0.31.0
-	k8s.io/client-go v0.31.0
-	k8s.io/kube-openapi v0.0.0-20240816214639-573285566f34
-	sigs.k8s.io/controller-runtime v0.19.0
+	github.com/onsi/ginkgo/v2 v2.22.2
+	github.com/onsi/gomega v1.36.2
+	github.com/prometheus/client_golang v1.22.0-rc.0
+	github.com/spf13/cobra v1.9.1
+	github.com/stretchr/testify v1.10.0
+	go.uber.org/zap v1.27.0
+	golang.org/x/exp v0.0.0-20240909161429-701f63a606c0
+	golang.org/x/time v0.11.0
+	google.golang.org/genproto/googleapis/api v0.0.0-20241209162323-e6fa225c2576
+	k8s.io/api v0.33.0-beta.0
+	k8s.io/apiextensions-apiserver v0.33.0-beta.0
+	k8s.io/apimachinery v0.33.0-beta.0
+	k8s.io/apiserver v0.33.0-beta.0
+	k8s.io/client-go v0.33.0-beta.0
+	k8s.io/kube-openapi v0.0.0-20250304201544-e5f78fe3ede9
+	sigs.k8s.io/controller-runtime v0.20.1-0.20250313041745-d82dcd80ca96
```

Comment:
Likely want to roll this back to a stable release

Reply:
Both client-go and controller-runtime are bumped to pre-release to get access to https://github.com/kubernetes-sigs/controller-runtime/pull/3135 (and the follow-up fix 3137); they should be released in v0.20.4 shortly, and then we should use that.